internal/cmd/asyncapi/command_asyncapi_test_utils.go

```diff
@@ -0,0 +1,222 @@
+				}, nil, nil
+			},
+			GetSubjectLevelConfigFunc: func(_ context.Context, _ string, _ *srsdk.GetSubjectLevelConfigOpts) (srsdk.Config, *http.Response, error) {
+				return srsdk.Config{CompatibilityLevel: BackwardCompatibilityLevel}, nil, nil
+			},
+			GetTopLevelConfigFunc: func(ctx context.Context) (srsdk.Config, *http.Response, error) {
+				return srsdk.Config{CompatibilityLevel: BackwardCompatibilityLevel}, nil, nil
+			},
+			GetTagsFunc: func(_ context.Context, _, _ string) ([]srsdk.TagResponse, *http.Response, error) {
+				return []srsdk.TagResponse{
+					{
+						TypeName: "Public",
+					},
+				}, nil, nil
+			},
+			GetTagDefByNameFunc: func(_ context.Context, _ string) (srsdk.TagDef, *http.Response, error) {
+				return srsdk.TagDef{Name: "Public", Description: "Public tag"}, nil, nil
+			},
+		},
+	},
+}
+
+func newCmd() (*command, error) {
+	cfg := &v1.Config{
+		BaseConfig: &config.BaseConfig{},
+		Contexts: map[string]*v1.Context{
+			"asyncapi": {
+				PlatformName: "confluent.cloud",
+				Credential:   &v1.Credential{CredentialType: v1.Username},
+				State: &v1.ContextState{
+					Auth: &v1.AuthConfig{
+						Organization: testserver.RegularOrg,
+						Account:      &ccloudv1.Account{Id: "env-asyncapi", Name: "asyncapi"},
+					},
+					AuthToken: "env-asyncapi",
+				},
+				SchemaRegistryClusters: map[string]*v1.SchemaRegistryCluster{
+					"lsrc-asyncapi": {
+						Id:                     "lsrc-asyncapi",
+						SchemaRegistryEndpoint: "schema-registry-endpoint",
+						SrCredentials:          &v1.APIKeyPair{Key: "ASYNCAPIKEY", Secret: "ASYNCAPISECRET"},
+					},
+				},
+				KafkaClusterContext: &v1.KafkaClusterContext{
+					EnvContext:         false,
+					ActiveKafkaCluster: "lkc-asyncapi",
+					KafkaClusterConfigs: map[string]*v1.KafkaClusterConfig{
+						"lkc-asyncapi": {
+							ID:           "lkc-asyncapi",
+							Name:         "AsyncAPI Cluster",
+							Bootstrap:    "kafka-endpoint",
+							RestEndpoint: "kafka-endpoint",
+							APIKeys: map[string]*v1.APIKeyPair{
+								"AsyncAPI": {Key: "ASYNCAPIKEY", Secret: "ASYNCAPISECRET"},
+							},
+							APIKey:     "AsyncAPI",
+							LastUpdate: time.Now(),
+						},
+					},
+				},
+			},
+		},
+		CurrentContext: "asyncapi",
+	}
+	prerunner := &pcmd.PreRun{Config: cfg}
+	cmd := new(cobra.Command)
+	c := &command{AuthenticatedStateFlagCommand: pcmd.NewAuthenticatedStateFlagCommand(cmd, prerunner)}
+	c.Command.Flags().String("resource", "lsrc-asyncapi", "resource flag for SR testing")
+	c.Version = &version.Version{Version: "1", UserAgent: "asyncapi"}
+	c.Command.Flags().String("schema-registry-api-key", "ASYNCAPIKEY", "API Key for schema registry")
+	c.Command.Flags().String("schema-registry-api-secret", "ASYNCAPISECRET", "API Secret for Schema Registry")
+	err := c.Command.Flags().Set("schema-registry-api-key", "ASYNCAPIKEY")
```

Comment:
We talked about this in your last PR - there's no need to mock an entire command. We have an integration test framework that can do that in a single line over in the test/ directory.

Reply:
Bump on this comment.